obspy/io/resp/nrl.py

```diff
@@ -178,20 +183,35 @@ def sensor_from_short(self, shortname):
         """
         return self._read_resp(self.sensor_path_from_short(shortname))
 
-    def parse_entire_library(self, path=None):
-        # Parse the entire NRL, for testing, or building another index.
-        if path is None:
-            path = self.join(self.root, self.index)
-        cp = self.read_ini(path)
+    def _recursive_parse(self, path):
+        data = {}
+        cp = self._read_ini(path)
         for section in cp.sections():
+            if section == 'Main':
+                continue
             for options in cp.options(section):
                 if 'path' in options:
-                    self.parse_entire_library(self.choose(section, path))
+                    data[section] = self._recursive_parse(
+                        self.choose(section, path))
+                    # XXX is it safe to assume that if "path" is present we
+                    # just follow down to the next level?
+                    continue
                 elif 'resp' in options:
-                    print(cp.get(section, 'resp'))
-                    pass
-                elif 'question' in options:
-                    print(cp.get(section, 'question'))
+                    data[section] = self._join(path, cp.get(section, 'resp'))
+                    continue
+                # XXX elif 'question' in options:
+                # XXX     print(cp.get(section, 'question'))
+        return data
+
+    def parse_entire_library(self):
```

Comment:
I did this as well once. I threw it out as I didn't want features available in the local and not the online. I do have an interactive tool for finding resps by answering questions and seeing answers, however not as a dict, perhaps I can do it that way(perhaps it can be done on the fly with getattr if that's the interface folks like). I think there was another reason I choose not to parse the entire nrl for use, but it's eluding me.